libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -111,7 +111,7 @@ bool NavEKF3_core::getRangeBeaconDebug(uint8_t &ID, float &rng, float &innov, fl
 bool NavEKF3_core::getHeightControlLimit(float &height) const
 {
     // only ask for limiting if we are doing optical flow navigation
-    if (frontend->_fusionModeGPS == 3) {
+    if (frontend->_fusionModeGPS == 3 && (PV_AidingMode == AID_RELATIVE) && flowDataValid) {
```

Comment:
Unrelated patch for flow?  Might be correct, just unrelated?

Reply:
It was carried across from EKF2 mods. It prevents the optical flow height limit being incorrectly applied applied when the  user has inhibited GPS use via the EK2_GPS_TYPE parameter and the vehicle is using either odometry or external nav measurements to navigate.